[AV2-1271] Update notification when chat message get deleted

sasangachathumal opened the pull request:
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/AV2-1271

**Additional Comments:**
`Add here`

**Checklist:**

- [x] Unit test completed?: (Y/N)
- [x] Docs folder up to date?: (Y/N)
- [x] Add if anything missed: (Y/N)

All checklists are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[AV2-1271]
updated notification service to use notification$ subject for chat.
remove newMessage$ notification subject.
added two new methods to notification service.
one for add chat notifications to notifications array.
one to remove chat notification from notification array.
remove newMessage$ subscriptions from tabs page.
remove newMessage$ subscriptions from notification page.
calling notification service after messages mark as read.

sasangachathumal committed:
Merge branch 'trunk-v3'

trtshen commented on projects/v3/src/app/pages/chat/chat-room/chat-room.component.ts:
```diff
@@ -353,6 +355,8 @@ export class ChatRoomComponent implements OnInit {
               readcount: messageIds.length
             });
+            // after messages read need to update chat notification data on notification service
+            this.notificationsService.getChatMessage().subscribe();
```
hi @sasangachathumal this one maybe can move to 'chat-badge-update' listener, or add another listener at a page where the notifications are gathered. what do you think?

because the line 359 here, look just like another broadcaster, which it means we have 2 broadcaster "chat-badge-update" and "getChatMessage" to force notification update. let's discuss tomorrow.

sasangachathumal committed:
[AV2-1771]
Remove notification service update from chat room component.
add event in notification service to update chat notifications.

sasangachathumal committed:
[Av2-1271]
Remove notification service from chat room component.
